test/test_linalg.py

```diff
@@ -2064,6 +2064,105 @@ def run_test_singular_input(batch_dim, n):
         for params in [(1, 0), (2, 0), (2, 1), (4, 0), (4, 2), (10, 2)]:
             run_test_singular_input(*params)
 
+    @precisionOverride({torch.float32: 1e-3, torch.complex64: 1e-3, torch.float64: 1e-7, torch.complex128: 1e-7})
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @dtypes(torch.float32, torch.float64, torch.complex64, torch.complex128)
+    def test_pinv(self, device, dtype):
+        from torch.testing._internal.common_utils import random_hermitian_pd_matrix
+
+        def run_test_main(A, hermitian):
+            # Testing against definition for pseudo-inverses
+            A_pinv = torch.linalg.pinv(A, hermitian=hermitian)
+            if A.numel() > 0:
+                self.assertEqual(A, A @ A_pinv @ A, atol=self.precision, rtol=self.precision)
+                self.assertEqual(A_pinv, A_pinv @ A @ A_pinv, atol=self.precision, rtol=self.precision)
```

Comment:
These lines probably want rtol = 0

Reply:
Mathematically yes, but it doesn't work with rtol=0. It fails for fp32 and complex64 and a larger rtol is needed for this test to pass. We do two matrix-matrix multiplications (`A @ A_pinv @ A`), which's really bad for fp32 if we compare element-wise. I think if compared in matrix norm these two objects would be very close to each other.

Reply:
OK, no worries.

Reply:
Note to self: we should support setting rtol and atol with the PrecisionOverride decorator.